src/mlpack/tests/smorms3_test.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file smorms3_test.cpp
+ * @author Vivek Pal
+ *
+ * Tests the SMORMS3 optimizer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
+#include <mlpack/core/optimizers/sgd/test_function.hpp>
+#include <mlpack/methods/logistic_regression/logistic_regression.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace arma;
+using namespace mlpack::optimization;
+using namespace mlpack::optimization::test;
+
+using namespace mlpack::distribution;
+using namespace mlpack::regression;
+
+using namespace mlpack;
+
+BOOST_AUTO_TEST_SUITE(SMORMS3Test);
+
+/**
+ * Tests the SMORMS3 optimizer using a simple test function.
+ */
+BOOST_AUTO_TEST_CASE(SimpleSMORMS3TestFunction)
+{
+  SGDTestFunction f;
+  SMORMS3Update smorms3Update(1e-16);
+  SGD<SGDTestFunction, SMORMS3Update> s(f, 1e-3, 5000000, 1e-9, true, smorms3Update);
```

Comment:
I would use SMORMS3 instead of ``SMORMS3Update smorms3Update(1e-16);`` in combination with the ``SGD<SGDTestFunction, SMORMS3Update>`` that was the idea of introducing a new class.